### PR TITLE
Update all major deps, change "master" branch to "main"

### PR DESCRIPTION
**Pull request by crutchcorn**
This PR:
- Updates all deps to their latest major release
- Migrates from some deprecated `metalsmith` packages to their new directory
- Moves the main `plop` branch from `master` to `main`, reflecting the main repo now (once this is merged, `master` will be deleted)

I'd also like to rename the main branch on this repo from `master` to `main`

Some things to note as part of this change:
- No styling changes or functionality changes have occurred (as far as I'm aware)
- Partials no longer are prefixed by their folder name, only the file name matters
- `node-fetch` required migration to ESM, while older `metalsmith` logic requires CJS. This is the reason for various `package.json` files present throughout to handle the switching back-and-forth

### CODE REVIEW

**Review by amwmedia — APPROVED** [review]
I think this is ready to go 👍🏻

File: helpers/basename.js [hunk 1]
```diff
@@ -1,5 +1,3 @@
-const {basename} = require('path');
-
 module.exports = function (path) {
-	return basename(path).split('.')[0];
+  return path.split(/[\/\.]/g)[0];
```
Comment (crutchcorn) [inline helpers/basename.js:2]: This still breaks on Windows for me (which is my main development environment). Could we potentially change this back to the `const {basename} = require('path')` or:

```suggestion
  return path.split(/[\/\.\\]/g)[0];
```

**Comment by crutchcorn** [comment]
Had one last fix for Windows environments (which shouldn't impact Linux/macOS environments), but LGTM

**Comment by crutchcorn** [comment]
@amwmedia can we also change the branch name on this repo from `master` to `main`? :) I figure I'll leave to you since I don't wanna accidentally bork Netlify

**Pull request merged**